src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/DefaultLSPEditorService.cs

```diff
@@ -64,21 +72,32 @@ public DefaultLSPEditorService(JoinableTaskContext joinableTaskContext, SVsServi
 
             await _joinableTaskFactory.SwitchToMainThreadAsync();
 
-            ApplyTextEdits(textEdits, snapshot, snapshot.TextBuffer);
+            var undoManager = _undoManagerProvider.GetTextBufferUndoManager(snapshot.TextBuffer);
+            var undoHistory = undoManager.TextBufferUndoHistory;
+            using var transaction = undoHistory.CreateTransaction("Apply TextEdits");
```

Comment:
Not sure what this text is for specifically but it'd probably be best to label it in a way that can be backtracked to us.

```suggestion
            using var transaction = undoHistory.CreateTransaction("Apply Razor LSP text edits");
```

Reply:
Sure. As far as I investigated, this is just for cosmetic purposes.